ts/src/hashkey.ts

```diff
@@ -0,0 +1,4313 @@
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {int} [params.fromId] the id of the entry to start from
+         * @param {int} [params.endId] the id of the entry to end with
+         * @returns {object[]} a list of [funding rate structures]{@link https://docs.ccxt.com/#/?id=funding-rate-history-structure}
+         */
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchFundingRateHistory() requires a symbol argument');
+        }
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetApiV1FuturesHistoryFundingRate (this.extend (request, params));
+        //
+        //     [
+        //         {
+        //             "id": "10698",
+        //             "symbol": "ETHUSDT-PERPETUAL",
+        //             "settleTime": "1722268800000",
+        //             "settleRate": "0.0001"
+        //         },
+        //         ...
+        //     ]
+        //
+        const rates = [];
+        for (let i = 0; i < response.length; i++) {
+            const entry = response[i];
+            const timestamp = this.safeInteger (entry, 'settleTime');
+            rates.push ({
+                'info': entry,
+                'symbol': this.safeSymbol (this.safeString (entry, 'symbol'), market, undefined, 'swap'),
+                'fundingRate': this.safeNumber (entry, 'settleRate'),
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+            });
+        }
+        const sorted = this.sortBy (rates, 'timestamp');
+        return this.filterBySinceLimit (sorted, since, limit) as FundingRateHistory[];
+    }
+
+    async fetchPositions (symbols: Strings = undefined, params = {}): Promise<Position[]> {
+        /**
+         * @method
+         * @description fetch open positions for a market
+         * @name hashkey#fetchPositions
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/get-futures-positions
+         * @description fetch all open positions
+         * @param {string[]|undefined} symbols list of unified market symbols
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.side] 'LONG' or 'SHORT' - the direction of the position (if not provided, positions for both sides will be returned)
+         * @returns {object[]} a list of [position structure]{@link https://docs.ccxt.com/#/?id=position-structure}
+         */
+        const methodName = 'fetchPositions';
+        if ((symbols === undefined) || (symbols.length < 1)) {
```

Comment:
`symbols.length` should be in an isolated statement like `const symbolsLength = symbols.length` otherwise will transpile as str_len in php